src/Functions/MatchImpl.h

```diff
@@ -391,22 +394,248 @@ struct MatchImpl
 
                 /// Tail, in which there can be no substring.
                 if (i < res.size())
-                    memset(&res[i], revert, (res.size() - i) * sizeof(res[0]));
+                    memset(&res[i], negate, (res.size() - i) * sizeof(res[0]));
             }
         }
     }
 
-    template <typename... Args>
-    static void vectorVector(Args &&...)
+    static void vectorVector(
+        const ColumnString::Chars & haystack_data,
+        const ColumnString::Offsets & haystack_offsets,
+        const ColumnString::Chars & needle_data,
+        const ColumnString::Offsets & needle_offset,
+        const ColumnPtr & start_pos_,
+        PaddedPODArray<UInt8> & res)
     {
-        throw Exception(ErrorCodes::ILLEGAL_COLUMN, "Function '{}' doesn't support non-constant needle argument", name);
+        const size_t haystack_size = haystack_offsets.size();
+
+        if (haystack_size != needle_offset.size())
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                    "Function '{}' unexpectedly received a different number of haystacks and needles", name);
+
+        if (start_pos_ != nullptr)
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                    "Function '{}' doesn't support start_pos argument", name);
```

Comment:
This checks looks more like LOGICAL_ERROR. Columns size should be equal (it's a caller invariant), argument types should have been checked before.

Reply:
True, fixed.